Packet++/src/WireGuardLayer.cpp

```diff
@@ -0,0 +1,151 @@
+#define LOG_MODULE PacketLogModuleWireGuardLayer
+
+#include "UdpLayer.h"
+#include "IPv4Layer.h"
+#include "IPv6Layer.h"
+#include "WireGuardLayer.h"
+#include "Logger.h"
+#include <sstream>
+#include <iomanip>
+#include <cstring>
+
+namespace pcpp
+{
+	std::string WireGuardLayer::toString() const
+	{
+		std::stringstream ss;
+
+		if (m_DataLen < sizeof(wg_common_header))
+		{
+			ss << "WireGuard header (incomplete)";
+			return ss.str();
```

Comment:
Generally a `stringstream::str()` allocates a new string buffer for the returned sting.

Since the return value if this branch is taken is constant `"WireGuard header (incomplete)"`, it would be better to return it directly to avoid allocating twice (once for `stringstream` and once for the return `string`).

`stringstream`'s declaration can also be moved below the current branch to avoid initializing it altogether if the branch is taken.

Reply:
Thanks for your feedback. I’ve removed the unused includes and refactored the stringstream usage to avoid unnecessary allocations.